api/src/main/java/org/apache/iceberg/exceptions/DuplicateFileException.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.exceptions;
+
+import java.util.List;
+import org.apache.iceberg.relocated.com.google.common.base.Joiner;
+
+/**
+ * This exception indicates that duplicate data files will be added to a table.
+ */
+public class DuplicateFileException extends RuntimeException {
```

Comment:
I don't have strong feelings about this but I don't think we need a new exception just for this, I think a Precondition.checkArguement or checkState would be fine with the appropriate message which you could keep as a constant string template in SparkTableUtil.

Reply:
Replaced with Preconditions.checkState